std/assembly/polyfills.ts

```diff
@@ -28,9 +28,9 @@ export function bswap<T extends number>(value: T): T {
 export function bswap16<T extends number>(value: T): T {
   if (isInteger<T>() && sizeof<T>() <= 4) {
     if (sizeof<T>() == 2) {
-      return <T>((value << 8) | ((value >> 8) & <T>0x00FF));
+      return <T>((value << 8) | ((value >>> 8) & <T>0x00FF));
     } else if (sizeof<T>() == 4) {
-      return <T>(((value << 8) & <T>0xFF00) | ((value >> 8) & <T>0x00FF) | (value & <T>0xFFFF0000));
+      return <T>(((value << 8) & <T>0xFF00) | ((value >>> 8) & <T>0x00FF) | (value & <T>0xFFFF0000));
```

Comment:
Wondering a bit about the usefulness of `bswap16`, since we are not using it anywhere. What are typical use cases for it?

Reply:
I don't know but post-MVP suggest add it: https://github.com/WebAssembly/design/blob/master/FutureFeatures.md#additional-integer-operators

Reply:
I see, hmm. If I'm not mistaken, semantics there seem a little different tho in that high bits are discarded.

Reply:
yes. `bswap16` preserve high part of value while `bswap<u16>` don't

Reply:
Do you think it would be possible to integrate it into `bswap`, as `bswap<u16>`, given the assumption that discarding high bits is fine?

Reply:
I think we could remove swap16 at least for now. I don't remind where it needed. Anyway this easily implement in user space if it really required in some project. wdyr?

Reply:
Yeah, I guess we could do this similar to other generic builtins. For instance, `load` and `load16_s` also become just a generic `load<T>`, and I'd expect `bswap` and `bswap16` to become a single `bswap<T>` as well. On `bswap<i16>`, the input is already known to have garbage in the high bits, so doing either just a `bswap16` on unsigned or a `bswap16` + sign-extend on signed seems sufficient.